torch_tvm/memory_utils.cpp

```diff
@@ -0,0 +1,51 @@
+#include "memory_utils.h"
+
+namespace torch_tvm {
+namespace utils {
+
+bool is_aligned(void* data_ptr, std::uintptr_t alignment_in_bytes) {
+  auto mask = alignment_in_bytes - 1;
+  AT_CHECK((alignment_in_bytes & mask) == 0);
+  return (reinterpret_cast<std::uintptr_t>(data_ptr) & mask) == 0;
+}
+
+DLManagedTensor* alloc_and_copy_data(const at::Tensor& tensor) {
+  DLManagedTensor* dl_managed_tensor = new DLManagedTensor();
+  // managed_tensor_deleter is supplied to unique_ptr as a deleter
+  // of this managed memory. Thus setting deleter to nullptr;
+  dl_managed_tensor->deleter = nullptr;
```

Comment:
why are we hijacking the lifetime logic? can we hand over a deleter that works with DLManagedTensor?  we don't want to have to maintain this lifetime stuff

Reply:
Read my comment above. I think we will have to still maintain the cache of some sort. If we dont want to manage lifetime we can hand it over to NDArray the cache will require to map Value* to NDArray*. If we agree on that, I think that is doable.